### PR TITLE
ODATA-1507 - Add test case for Filter Path Segment followed by key predicate

### PR DESCRIPTION
**Pull request by ralfhandl**
`filterExpr` can already be reached via `collectionPathExpr`, so this rule alternative is redundant

### CODE REVIEW

**Review by sthagen — APPROVED** [review]

File: abnf/odata-abnf-construction-rules.txt [hunk 1]
```diff
@@ -549,7 +549,7 @@ collectionNavigationExpr = collectionNavNoCastExpr
 
 collectionNavNoCastExpr = collectionPathExpr
                         / keyPredicate [ singleNavigationExpr ] 
-                        / filterExpr [ collectionNavigationExpr ]                     
+                      ;  / filterExpr [ collectionNavigationExpr ]
```
Comment (HeikoTheissen) [inline abnf/odata-abnf-construction-rules.txt:552]: Removing the rule removes the option `collectionNavNoCastExpr = filterExpr keyPredicate`, therefore it is not redundant.

**Review by HeikoTheissen — CHANGES_REQUESTED** [review]

File: abnf/odata-abnf-construction-rules.txt (hunk 1 above)
Comment (ralfhandl) [reply abnf/odata-abnf-construction-rules.txt:552]: A key predicate after a filter does not make sense: just use the key predicate directly, so I would few this as an added benefit 😄

Comment (HeikoTheissen) [reply abnf/odata-abnf-construction-rules.txt]: Technically, it is an incompatible (if sensible) change to the ABNF that deserves discussion through an OASIS issue.

Comment (HeikoTheissen) [reply abnf/odata-abnf-construction-rules.txt]: See [ODATA-1507](https://issues.oasis-open.org/browse/ODATA-1507).

Comment (ralfhandl) [reply abnf/odata-abnf-construction-rules.txt]: Thanks, added to the TC agenda

**Review by sthagen — APPROVED** [review]

File: abnf/odata-abnf-construction-rules.txt (hunk 1 above)
Comment (HeikoTheissen) [reply abnf/odata-abnf-construction-rules.txt]: TC 2022-03-10 agreed to keep the option `collectionNavNoCastExpr = filterExpr keyPredicate`. I have added a test case, this proves that the rule is not redundant.

**Comment by HeikoTheissen** [comment]
> The parser generator we use requires to order alternatives by "best match" or "most distinctive", so the rather distinctive `/$filter(...)` has to go before the less distinctive collection path expression.

But it can happen that one test case demands one ordering and another test case demands a different ordering, right?

**Review by HeikoTheissen — APPROVED** [review]

**Comment by ralfhandl** [comment]
In that case we can add constraints to force the parser skip an alternative, for example by telling it that `$filter(foo)` is not a key value.

**Review by sthagen — APPROVED** [review]

**Pull request merged**